File: server/src/protocol/BlindsStatusResponse.cpp
#include "BlindsStatusResponse.h"
#include "ResponseVisitor.h"

using namespace std;
using namespace communication;

BlindsStatusResponse::BlindsStatusResponse(BlindsStatus status, BlindsMode mode)
        : Response(10, ResponseType::BLINDS_STATUS_RSP),
          status_(status),
          mode_(mode)

{}

BlindsStatusResponse::~BlindsStatusResponse()
{}

vector<uint8_t> BlindsStatusResponse::getFrameBytes()
{
    initializeDataSize();

    vector<uint8_t > frame = getHeader();
    frame.push_back(static_cast<uint8_t>(responseType_));
    frame.push_back(static_cast<uint8_t>(status_));
    frame.push_back(static_cast<uint8_t>(mode_));

    return frame;
}

std::string BlindsStatusResponse::getName()
{
    return string("BlindsStatusResponse");
}

void BlindsStatusResponse::accept(ResponseVisitor &visitor)
{
    visitor.visit(*this);
}

BlindsStatus BlindsStatusResponse::getBlindsStatus() const
{
    return status_;
}

void BlindsStatusResponse::setBlindsStatus(BlindsStatus status)
{
    status_ = status;
}

BlindsMode BlindsStatusResponse::getBlindsMode() const
{
    return mode_;
}

void BlindsStatusResponse::setBlindsMode(BlindsMode mode)
{
    mode_ = mode;
}

void BlindsStatusResponse::initializeDataSize()
{
    uint16_t dataSize = sizeof(responseType_);
    dataSize += sizeof(status_);
    dataSize += sizeof(mode_);

    setDataSize(dataSize);
}

File: mobile_client/app/src/main/java/wpam/mobile_client/protocol/BlindsStatusResponse.java
package wpam.mobile_client.protocol;

import java.util.ArrayList;

public final class BlindsStatusResponse extends Response
{
	private BlindsStatus status_;
	private BlindsMode mode_;

	public BlindsStatusResponse(BlindsStatus status, BlindsMode mode)

	{
		super(10, ResponseType.BLINDS_STATUS_RSP);
		this.status_ = status;
		this.mode_ = mode;
	}
	public final void close()
	{
		super.close();
	}

	@Override
	public ArrayList<Integer> getFrameBytes()
	{
		initializeDataSize();

		ArrayList<Integer> frame = getHeader();
		frame.add(responseType_.getValue());
		frame.add(status_.getValue());
		frame.add(mode_.getValue());

		return frame;
	}

	@Override
	public String getName()
	{
		return new String("BlindsStatusResponse");
	}

	@Override
	public void accept(ResponseVisitor visitor)
	{
		visitor.visit(this);
	}

	public BlindsStatus getBlindsStatus()
	{
		return this.status_;
	}

	public void setBlindsStatus(BlindsStatus status)
	{
		this.status_ = status;
	}

	public BlindsMode getBlindsMode()
	{
		return this.mode_;
	}
	public void setBlindsMode(BlindsMode mode)
	{
		this.mode_ = mode;
	}

	@Override
	protected void initializeDataSize()
	{
		setDataSize(3);
	}
}